File: leetcode/2028-find-missing-observations/missingRolls.ts
function missingRolls(rolls: number[], mean: number, n: number): number[] {
  const result: number[] = [];
  let missingSum = mean * (rolls.length + n);
  for (const roll of rolls) {
    missingSum -= roll;
  }

  const avg = Math.floor(missingSum / n);
  const rest = missingSum % n;

  if (missingSum >= n && missingSum <= 6 * n) {
    for (let i = 1; i <= n; i++) {
      if (i <= rest) {
        result[i - 1] = avg + 1;
      } else {
        result[i - 1] = avg;
      }
    }
  }

  return result;
}

export default missingRolls;

File: leetcode/2028-find-missing-observations/missingRolls.go
package leetcode

func missingRolls(rolls []int, mean int, n int) []int {
	var result = []int{}
	var missingSum = mean * (n + len(rolls))
	for _, roll := range rolls {
		missingSum = missingSum - roll
	}

	if missingSum >= n && missingSum <= 6*n {
		var avg = missingSum / n
		var rest = missingSum % n

		for i := 1; i <= n; i++ {
			if i <= rest {
				result = append(result, avg+1)
			} else {
				result = append(result, avg)
			}
		}
	}

	return result
}
